### ISSUE TITLE
Allow bots

### ISSUE THREAD

**Issue by amyxzhang** [haystack/murmur#86]
Would be cool to allow a bot hook in so that people can add and implement email bots to their group. Then you can call the bot via email to reply in a thread.

**Comment by karger** [comment]
This may be the tip of an iceberg, as it kind of suggests "build a 
server for arbitrary types of communication", i.e. something that could 
serve for chat, twitter, etc.  The point being that we are demonstrating 
the value of building a *back end server* whose primary interface is its 
*api*, as opposed a *site* (twitter, snapchat, instagram, whatsapp) 
which locks you in to a particular front end in order to use the back 
end.  This would be far better for consumers who could choose from (and 
even create) a variety of front ends to access the service in a variety 
of ways.  Email is a beautiful example of this----think about the 
diversity of email clients compared to the diversity of facebook 
clients.  Everybody can pick one they like.

This will never fly in the for-profit world, because the front end is 
where you stick your advertisements and make your money.  But it would 
be awsome for society.

**Comment by amyxzhang** [comment]
Yep, this is how I've been envisioning murmur! We will write the hook-ins
to a lot of those applications (for instance I think bridging the
synchronous/asynchronous gap will need some alterations to the discussion)
 but certainly it would be great if other people wanted to write them too.

The bot API I thought of just from looking at the popularity of Slack bots
(and Twitter bots and soon FB messenger bots). You could call a giphy bot
to chime in to a conversation or hook-ins to post when there are automatic
notifications (each with their own tag so you can mute/divert) (one of
which could be the papers survey for instance!)

On Tue, Apr 26, 2016 at 6:17 PM, David Karger <notifications@github.com>
wrote:

> This may be the tip of an iceberg, as it kind of suggests "build a
> server for arbitrary types of communication", i.e. something that could
> serve for chat, twitter, etc. The point being that we are demonstrating
> the value of building a *back end server* whose primary interface is its
> *api*, as opposed a *site* (twitter, snapchat, instagram, whatsapp)
> which locks you in to a particular front end in order to use the back
> end. This would be far better for consumers who could choose from (and
> even create) a variety of front ends to access the service in a variety
> of ways. Email is a beautiful example of this----think about the
> diversity of email clients compared to the diversity of facebook
> clients. Everybody can pick one they like.
>
> This will never fly in the for-profit world, because the front end is
> where you stick your advertisements and make your money. But it would
> be awsome for society.
>
> On 4/26/2016 5:12 PM, Amy Zhang wrote:
> >
> > Would be cool to allow a bot hook in so that people can add and
> > implement email bots to their group. Then you can call the bot via
> > email to reply in a thread.
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly or view it on GitHub
> > <https://github.com/haystack/murmur/issues/86>
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/haystack/murmur/issues/86#issuecomment-214904803>
>



-- 
Amy X. Zhang | Ph.D. student at MIT CSAIL | http://people.csail.mit.edu/axz
| @amyxzh
